x-pack/legacy/plugins/siem/public/components/super_date_picker/index.tsx

```diff
@@ -139,27 +145,30 @@ export const SuperDatePickerComponent = class extends Component<
     );
   }
   private onRefresh = ({ start, end, refreshInterval }: OnRefreshProps): void => {
-    this.updateReduxTime({
-      start,
+    const { kqlHaveBeenUpdated } = this.props.updateReduxTime({
       end,
-      isQuickSelection: this.state.isQuickSelection,
+      id: this.props.id,
       isInvalid: false,
+      isQuickSelection: this.state.isQuickSelection,
+      kql: this.props.queries.find(q => q.id === 'kql') as inputsModel.GlobalKqlQuery,
```

Comment:
optional: since the `kql` ID is special and referred to in other parts of the code, such as [refeshQuery](https://github.com/elastic/kibana/pull/45065/files#diff-85bc39e81b9c63122f4cb9406f223606R192), [onTimeChange](https://github.com/elastic/kibana/pull/45065/files#diff-85bc39e81b9c63122f4cb9406f223606R204), [componentDidUpdate](https://github.com/elastic/kibana/pull/45065/files#diff-85bc39e81b9c63122f4cb9406f223606R204), etc  consider creating a `const` for it and replacing the usages of `'kql'` with the `const`